sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,246 @@
+local tostring = tostring
+local type = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+local function influxdb_kv_fmt(string)
+    return tostring(string):gsub("([ ,])", "\\%1")
+end
+
+local function points_tags_tables(config)
+    local name_prefix = ""
+    if config["name_prefix"] then
+        name_prefix = interp.interpolate_from_msg(config["name_prefix"])
+    end
+    local name_prefix_delimiter = config["name_prefix_delimiter"] or ""
+    local used_tag_fields = config["used_tag_fields"]
+    local skip_fields = config["skip_fields"]
+
+    -- Initialize the tags table, including base field tag values in
+    -- list if the magic **all** or **all_base** config values are
+    -- defined.
+    local tags = {}
+    if not config["carbon_format"]
+        and (config["tag_fields_all"]
+        or config["tag_fields_all_base"]
+        or config["used_tag_fields"]) then
+            for field in pairs(field_util.base_fields_tag_list) do
+                if config["tag_fields_all"] or config["tag_fields_all_base"] or used_tag_fields[field] then
+                    local value = read_message(field)
+                    table.insert(tags, influxdb_kv_fmt(field).."="..tostring(influxdb_kv_fmt(value)))
+                end
+            end
+    end
+
+    -- Initialize the table of data points and populate it with data
+    -- from the Heka message.  When skip_fields includes "**all_base**",
+    -- only dynamic fields are included as InfluxDB data points, while
+    -- the base fields serve as tags for them. If skip_fields does not
+    -- define any base fields, they are added to the fields of each data
+    -- point and each dynamic field value is set as the "value" field.
+    -- Setting skip_fields to "**all_base**" is recommended to avoid
+    -- redundant data being stored in each data point (the base fields
+    -- as fields and as tags).
+    local points = {}
+    while true do
+        -- Iterate through Fields array in the message
+        local typ, name, value = read_next_field()
+
+        -- Exit the perpetual loop when the iteration is complete
+        if not typ then break end
+
+        -- Replace non-word characters with underscores for Carbon
+        -- to avoid periods resulting in extraneous directories
+        if config["carbon_format"] then
+            name = name:gsub("[^%w]", "_")
+        end
+
+        -- Include the dynamic fields as tags if they are defined in
+        -- configuration or the magic value "**all**" is defined.
+        -- Convert value to a string as this is required by the API
+        if not config["carbon_format"]
+            and (config["tag_fields_all"]
+            or (config["used_tag_fields"] and used_tag_fields[name])) then
+                table.insert(tags, influxdb_kv_fmt(name).."="..tostring(influxdb_kv_fmt(value)))
+        end
+
+        if config["source_value_field"] and name == config["source_value_field"] then
```

Comment:
This was changed from using `exclude_name` to the config value that replaced it (and makes it more flexible), `source_value_field`, which indicates which field in the message to pull the value from.